### DISCUSSION TITLE
Support for SoLiD-OIDC

### DISCUSSION THREAD

**Discussion by joeitu** [keycloak/keycloak#9150] [Ideas]
### Description

Hello,

Is keycloak planning to support Solid-OIDC 1?
If yes, when would Solid-OIDC be available to keycloak instance?

Best regards

### Motivation

I would allow keycloak users to log in to Solid Pod with their keycloak SSO account instead of having to create a new account on the Solid Pod instance.

**Comment by dteleguin** [comment]
This is currently work in progress. As Solid-OIDC is a big feature, we will be a proper discussion and likely a series of PRs. Stay tuned!

**Reply by samuk** [reply]
I'm interested in this. Is there any work happening around it at the moment?

**Reply by RaggedStaff** [reply]
Also interested in this... we're currently looking at implementing straight OIDC but with a view to moving to Solid-OIDC in the future. Is there a roadmap for the availability of this at all?

**Reply by samuk** [reply]
Not via Keycloak.

I am aware of https://activitypods.org/ which can act as an OIDC provider and is adding Solid auth at the moment. You also get activity-pub for 'free'

**Reply by dteleguin** [reply]
@samuk @RaggedStaff there's a PoC for Keycloak Solid-OIDC here: https://github.com/CarrettiPro/keycloak-solid
It's still a WIP and even lacks a README, but I'm planning to add it ASAP
I would appreciate any feedback, so don't hesitate to reach out to me here or on CNCF Slack (Dmitry Telegin)

**Reply by sotberd** [reply]
@dteleguin I'm also interested in this, is there any documentation on How it can be used with Solid-OIDC?

**Comment by naturzukunft** [comment]
Hi @dteleguin how is currently the state of your PoC?
br Fredy

**Comment by esteiner** [comment]
Hi @dteleguin 
I had a try with the code at https://github.com/CarrettiPro/keycloak-solid. What I see is that the current implementation is missing the dynamic client registration in the case that the "client_id" parameter of the authentication request is a URL (to the ClientID document). Is that correct?
